pyfar/dsp/dsp.py

```diff
@@ -187,3 +187,112 @@ def spectrogram(signal, dB=True, log_prefix=20, log_reference=1,
+
+    Parameters
+    ----------
+    signal: Signal
+        Input signal of the signal class
+    norm_type: string
+        'time' - Normalize the time signal to 'value'
+        'abs' - Normalize the magnitude spectrum to 'value'
+        'dB' - Normalize the db magnitude spectrum to 'value'
+        The default is 'time'
+    operation: string
+        'max' - Normalize to the absolute maximum of the signal data
+        'mean' - Normalize to the mean of the signal data
+        'rms' - Normalize to the rms of the signal data
+        The default is 'max'
+    channelwise: string
+        'each' - Normalize each channel separately
+        'max' - Normalize to the max or operation across all channels
+        'min' - Normalize to min or operation across all channels
+        'mean' - Normalize to mean of operation across all channels
+       The default is 'max'
+    value:
+        Normalizes to 'value' which can be a scalar or a vector with
+        a number of elements equal to channels
+        The default is 0 for norm_type='db' and 1 otherwise
+    freq_range: tuple
+        two element vector specifying upper and lower frequency bounds
+        for normalization or scalar specifying the centre frequency for
+        normalization
+        The default is (20,22000)
+    Returns
+    --------
+    normalized_signal: Signal
+        The normalized signal
+    """
+
+    # check input
+    if not isinstance(signal, Signal):
+        raise TypeError('Input data has to be of type: Signal.')
+
+    # set default values
+    if value is None:
+        value = 0 if norm_type == 'dB' else 1
+
+    # copy and transform data to the desired domain
+    if norm_type == 'time':
+        normalized_input = np.abs(signal.time.copy())
+    elif norm_type == 'abs':
+        normalized_input = np.abs(signal.freq.copy())
+    elif norm_type == 'dB':
+        normalized_input = 20 * np.log10(signal.freq.copy())
+    else:
+        raise ValueError(("norm _type must be 'time', 'abs' or 'dB'."))
+
+    # take logarithm of the data here instead of later?
+
+    # get bounds for normalization
+    if norm_type == 'time':
+        lim = (0, signal.n_samples)
+
+    else:
+        lim = signal.find_nearest_frequency(freq_range)
+        print(lim)
+
+        # remove 0 hz and nyquist due to normalization dependency
+        if signal.n_samples % 2:
+            lim[0] = np.max([lim[0, 1]])
+        else:
+            lim = np.clip(lim, 0)
+
+    # get values for normalization
+    if operation == 'max':
+        values = np.max(normalized_input[..., lim[0]:lim[1]], axis=-1,
+                        keepdims=True)
+    elif operation == 'mean':
+        values = np.mean(normalized_input[..., lim[0]:lim[1]], axis=-1,
+                         keepdims=True)
+    elif operation == 'rms':
+        values = np.sqrt(np.mean((normalized_input[..., lim[0]:lim[1]])**2,
+                         axis=-1, keepdims=True))
```

Comment:
- Add `axis` and `keepdims` also for `np.sqrt`
- Remove extra parentheses around `normalized_input...`